groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyLanguageServer.java

```diff
@@ -67,8 +69,10 @@ public GroovyLanguageServer(
     @Override
     public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
         logger.debug("Initializing Groovy Language Server");
-        workspaceRoot = Uris.getAbsolutePath(params.getRootPath());
-        logger.debug("Resolve workspace root from '{}' to '{}'", params.getRootPath(), workspaceRoot);
+        workspaceRoot = Paths.get(Optional.ofNullable(params.getRootUri())
+                .map(URI::create)
+                .orElseGet(() -> Paths.get(params.getRootPath()).toUri()));
```

Comment:
the exception message here can be unclear if both rootUri and rootPath do not exist in initializeParams.
Probably fine if we want to handle the use case that initializeParams are sent from the authoring